scripts/iam-tools/aws_api_key_manager.py

```diff
@@ -225,19 +225,25 @@ def get_token(aws_account):
                 if yaml_config["idp_host"]:
                     ops_idp_host = yaml_config["idp_host"]
 
-                creds = saml_aws_creds.get_temp_credentials(
-                    metadata_id='urn:amazon:webservices:%s' % aws_account,
-                    idp_host=ops_idp_host
-                    )
-
-                client = boto3.client(
-                    'iam',
-                    aws_access_key_id=creds['AccessKeyId'],
-                    aws_secret_access_key=creds['SecretAccessKey'],
-                    aws_session_token=creds['SessionToken']
-                    )
-
-                return client
+                try:
+                    creds = saml_aws_creds.get_temp_credentials(
+                        metadata_id='urn:amazon:webservices:%s' % aws_account,
+                        idp_host=ops_idp_host
+                        )
+
+                    client = boto3.client(
+                        'iam',
+                        aws_access_key_id=creds['AccessKeyId'],
+                        aws_secret_access_key=creds['SecretAccessKey'],
+                        aws_session_token=creds['SessionToken']
+                        )
+                    return client
+
+                except ValueError as client_exception:
+                    if 'Error retrieving SAML token' and 'Metadata not found'\
```

Comment:
this looks like a synonym of `if True and 'Metadata not found' in ...`
you probably meant:
` if 'Error retrieving SAML token' in client_exception.message \
  and 'Metadata not found' in client_exception.message`